src/plugins/expression-builder/where-schema.js

```diff
@@ -0,0 +1,298 @@
+import {typeMapping as operations} from './operations';
+import suggestFactory from './suggest';
+
+export function whereSchema(builder, $q, context) {
+	const {suggest, suggests} = suggestFactory($q, context)('#field');
+
+	return builder
+		.node('#logical', function (schema) {
+			schema
+				.attr('serialize', {
+					'#logical-op': ['value']
+				})
+				.select('#logical-op', {
+					classes: ['cb-operation'],
+					options: ['AND', 'OR'],
+					value: 'AND'
+				})
+				.iconButton('#add-logical', {
+					icon: 'add',
+					click: (node, line) => {
+						node.addChildAfter(node.clone());
+					}
+				})
+				.iconButton('#remove-logical', {
+					icon: 'close',
+					isVisible: node => node.level > 1,
+					click: node => {
+						node.remove();
+					}
+				})
+				.node('#condition', function (schema) {
+					schema
+						.attr('placeholder', true)
+						.attr('class', {
+							placeholder: node => node.attr('placeholder')
+						})
+						.attr('serialize', {
+							'#field': ['value'],
+							'#operator': ['value'],
+							'#value': ['value'],
+							'#from': ['value'],
+							'#to': ['value'],
+							'#in-operand': ['values'],
+							'@attr': ['placeholder']
+						})
+						.select('#field', {
+							classes: ['cb-operation', 'field'],
+							options: context.fields.map(function (f) {
+								return f.name;
+							}),
+							value: context.fields.length ? context.fields[0].name : '',
+							getLabel: (node, line, name) => {
+								const field = context.fields.filter(function (f) {
+									return f.name === name;
+								})[0];
+
+								return (field && field.label) || null
+							},
+							getType: (node, line, name) => {
+								const field = context.fields.filter(function (f) {
+									return f.name === name;
+								})[0];
+
+								return (field && field.type) || 'text';
+							},
+							change: function (node, line) {
+								if (node.attr('placeholder')) {
+									node.addAfter(node.clone());
+									node.attr('placeholder', false);
+								}
+
+								var field = this.value,
```

Comment:
Can ```const``` be used here?
Also formatting doesn't look pretty

Reply:
Yes, I'll use const. 
what's wrong about formatting?

Reply:
Cause of you write variables divided by comma, next row variable shifted to right by double tab. Cause of we're using tabs in the project, in github review it's located far from first variable row declaration. Really it's not a big problem, more my personal hating of tab formatting issues :)

Reply:
Yeah, agree, there's some pain about reviewing such structures.